crates/sui-verifier-transactional-tests/tests/entry_points/non_key_struct_generic.mvir

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// invalid as NoStore doesn't have store, so Obj doesn't have key
+
+//# publish
+module 0x0.M {
+    import 0x2.TxContext;
+    import 0x2.ID;
+
+    struct Obj<T> has key { id: ID.VersionedID }
```

Comment:
Just out of curiosity, does explicitly making `T` phantom changes things at all?

Reply:
Yes, this example would be allowed if `T` was phantom